### PR TITLE
root: 6.32.06 -> 6.32.08, fix on darwin, move to pkgs/by-name

### PR DESCRIPTION
**Pull request by veprbl**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 355981` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>herwig</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.rivet</li>
    <li>python312Packages.hepmc3</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>root</li>
    <li>root.debug</li>
    <li>sherpa</li>
    <li>thepeg</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Review by guitargeek — APPROVED** [review]
Thank you very much, looks great!

File: pkgs/by-name/ro/root5/package.nix [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
-, libGLU
-, libGL
-, zlib
-, libxml2
-, libxcrypt
-, lz4
-, xz
-, gsl
-, xxHash
-, Cocoa
```
[the remark this answers is not in the document]
Comment (veprbl) [reply pkgs/by-name/ro/root5/package.nix:22]: This is how new SDK works in nixpkgs [1], everything is put together in a similar way to it's done by Apple. The individual packages are now empty dummies and OpenGL in the cmake flag even breaks the build. I was in shock too :)

[1] https://discourse.nixos.org/t/the-darwin-sdks-have-been-updated/55295

[the remark this answers is not in the document]
Comment (guitargeek) [reply pkgs/by-name/ro/root5/package.nix:22]: Thanks for the explanation! I mean I was shocked in a positive way, this is quite nice!

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 355981` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>hepmc3</li>
    <li>herwig</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.rivet</li>
    <li>python312Packages.hepmc3</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>root</li>
    <li>root.debug</li>
    <li>sherpa</li>
    <li>thepeg</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 355981` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>apfelgrid</li>
    <li>applgrid</li>
    <li>root5</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>herwig</li>
    <li>sherpa</li>
    <li>thepeg</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.rivet</li>
    <li>python312Packages.hepmc3</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>root</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 355981` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>apfelgrid</li>
    <li>applgrid</li>
    <li>root5</li>
  </ul>
</details>
<details>
  <summary>13 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>herwig</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.rivet</li>
    <li>python312Packages.hepmc3</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>root</li>
    <li>root.debug</li>
    <li>sherpa</li>
    <li>thepeg</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 355981` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>16 packages built:</summary>
  <ul>
    <li>apfelgrid</li>
    <li>applgrid</li>
    <li>hepmc3</li>
    <li>herwig</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.rivet</li>
    <li>python312Packages.hepmc3</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>root</li>
    <li>root.debug</li>
    <li>root5</li>
    <li>sherpa</li>
    <li>thepeg</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 355981` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>apfelgrid</li>
    <li>applgrid</li>
    <li>root5</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>herwig</li>
    <li>sherpa</li>
    <li>thepeg</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.rivet</li>
    <li>python312Packages.hepmc3</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>root</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Comment by veprbl** [comment]
Result of `nixpkgs-review pr 355981` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages marked as broken and skipped:</summary>
  <ul>
    <li>apfelgrid</li>
    <li>applgrid</li>
    <li>root5</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>herwig</li>
    <li>sherpa</li>
    <li>thepeg</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>hepmc3</li>
    <li>ns-3</li>
    <li>python311Packages.hepmc3</li>
    <li>python311Packages.rivet</li>
    <li>python312Packages.hepmc3</li>
    <li>rivet (python312Packages.rivet)</li>
    <li>root</li>
    <li>xfitter</li>
    <li>yoda-with-root</li>
  </ul>
</details>

**Pull request merged**